### PR TITLE
User request improvements

### PR DESCRIPTION
**Pull request by tiferrei**
This adds all the requests from the `User` API call.

Tiago

### CODE REVIEW

**Comment by pietbrauer** [comment]
Could you please add tests as well?

**Comment by tiferrei** [comment]
It would be my pleasure, I just don't know which kind of tests you would like, as in, what to test? Because there are no tests for the `GET` requests yet, I really don't know how you'd like them.

**Comment by tiferrei** [comment]
BTW, I found an error on the `repositories` request that completely breaks the call, will fix it on the next pull request together with the name change of `repositories` to `projects`.

**Comment by pietbrauer** [comment]
Yes, please test the GET call too. You can have a look at the tests in OctoKit Tests https://github.com/nerdishbynature/octokit.swift/blob/master/OctoKitTests/UserTests.swift

**Comment by tiferrei** [comment]
Cool, thanks!!

**Comment by tiferrei** [comment]
@pietbrauer , there are some things I don't understand here:

``` swift
func testGettingAuthenticatedUser() {
        let session = OctoKitURLTestSession(expectedURL: "https://api.github.com/user?access_token=token", expectedHTTPMethod: "GET", jsonFile: "user_me", statusCode: 200)
        let task = Octokit(TokenConfiguration("token")).me(session) { response in
            switch response {
            case .Success(let user):
                XCTAssertEqual(user.login, "pietbrauer")
            case .Failure(let error):
                XCTAssert(false, "should not retrieve an error \(error)")
            }
        }
        XCTAssertNotNil(task)
        XCTAssertTrue(session.wasCalled)
    }
```
1. The token is blank on both the session and the API call, why is it supposed to get the user?
2. If the token is blank, how would the `login` from that call ever be equal to `pietbrauer`?

Thanks,
Tiago

**Comment by pietbrauer** [comment]
It gets a stubbed session that returns the JSON `user_me` from the Fixtures directory. I think it's best to check out the Octokit repository and open it in Xcode, set a breakpoint at the stuff you have problems understanding.

**Comment by tiferrei** [comment]
Ah yes alright, yup got it! Now I have a "Test User" on my gitlab server with some dummy content for when I am testing the API calls, can I use it for the fixtures or would you like to use another user in specific?

**Comment by pietbrauer** [comment]
Just use something.

**Comment by pietbrauer** [comment]
You can have a look at https://github.com/nerdishbynature/octokit.swift/pull/46 maybe this gives you a hint on why your build is failing.

**Comment by tiferrei** [comment]
Yeah, these seem to be the bugs:

```
❌  /Users/travis/build/nerdishbynature/TanukiKit/TanukiKitTests/UserTests.swift:11:13: constant 'task' inferred to have type '()', which may be unexpected
let task = TanukiKit(config).me(session) { response in
            ^
⚠️  /Users/travis/build/nerdishbynature/TanukiKit/TanukiKitTests/UserTests.swift:26:13: constant 'task' inferred to have type '()', which may be unexpected
let task = TanukiKit(config).me(session) { response in
            ^
⚠️  /Users/travis/build/nerdishbynature/TanukiKit/TanukiKitTests/UserTests.swift:41:13: constant 'task' inferred to have type '()', which may be unexpected
let task = TanukiKit().me(session) { response in
            ^
⚠️  /Users/travis/build/nerdishbynature/TanukiKit/TanukiKitTests/UserTests.swift:59:62: cast from 'String?' to unrelated type '[String : AnyObject]' always fails
let subject = User(TestHelper.stringFromFile("User") as! [String: AnyObject])
            ^
```

I just don't understand this: `constant 'task' inferred to have type '()', which may be unexpected` as the code is basically the same as `OctoKit` but for `TanukiKit`.

**Comment by pietbrauer** [comment]
It is not, that is why I send you the other pull request. TanukiKit does not return a task yet, which is why the resulting type is a function, `which may be unexpected`.

**Comment by tiferrei** [comment]
Oh yes! Of course, that was stupid, thanks!!

**Comment by tiferrei** [comment]
Done.

File: TanukiKit/User.swift [hunk 1]
```diff
@@ -3,20 +3,60 @@ import RequestKit
...
+    public var twoFactorEnabled: Bool?
+    public var external: Bool?
     public var privateToken: String?
 
     public init(_ json: [String: AnyObject]) {
         if let id = json["id"] as? Int {
+            name = json["name"] as? String
+            username = json["username"] as? String
             self.id = id
-            login = json["username"] as? String
```
Comment (pietbrauer) [inline TanukiKit/User.swift:16]: This property was changed from `login` to `name` and is a breaking change in the API, I would rather keep it without being a breaking change for now. Can you rename it to `login`, please?

File: TanukiKitTests/User.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (pietbrauer) [inline TanukiKitTests/User.json]: Can you move this into  `TanukiKitTests/Fixtures/User.json`, please?

**Comment by pietbrauer** [comment]
Looks good, just 2 💬 and could you please rebase this into a single commit? Twelve seem to be a bit too many for a relatively small change.

**Comment by tiferrei** [comment]
Yup, commit 12 applies your comments, will rebase into one commit.

**Comment by tiferrei** [comment]
Is that alright?

**Comment by tiferrei** [comment]
BTW, can you please comment on the other PR when you rebase into other branches?

**Comment by tiferrei** [comment]
@pietbrauer, we have a small problem, on both 3.0 and 2.3 branches, the podspec is still pointing to RequestKit 0.3.0, so it builds perfectly as source code, but when using TanukiKit in another project as a cocoa pod, it kinda breaks the Routers as the syntax changed a bit with the new RequestKit so it fails to compile. And to get things even worse, we can't specific branches for dependencies in the podspec, it only accepts releases.

For example, errors like:
Use of undeclared type 'RequestKitURLSession' and Type 'PublicKeyRouter' does not conform to protocol 'Router' as this version still required the params to be [String : String]

**Comment by tiferrei** [comment]
Well, that's awkward. My squashed commit actually squashed all the commits but it called some others?

**Comment by tiferrei** [comment]
Is there any way I can remove the previous ones and just add the Squashed?

**Comment by tiferrei** [comment]
I hate Git.

**Comment by tiferrei** [comment]
Is it OK if I close this PR, clean the mess and open another?

**Pull request closed without merging**